daml/Marketplace/Custody/Service.daml

```diff
@@ -77,62 +43,22 @@ template Service
           create Custody.LifecycleRequest with ..
 
     controller provider can
-      nonconsuming OpenAccount : ContractId AssetSettlementRule
-        with
-          openAccountRequestCid : ContractId Custody.OpenAccountRequest
-        do
-          Custody.OpenAccountRequest{accountId; observers; ctrls} <- fetchAndArchive openAccountRequestCid
 
-          let account = Account with provider; owner = customer; id = accountId
-          createOrLookup AssetSettlementRule with ..
-
-      nonconsuming OpenAllocationAccount : ContractId AllocationAccountRule
-        with
-          openAllocationAccountRequestCid : ContractId Custody.OpenAllocationAccountRequest
-        do
-          Custody.OpenAllocationAccountRequest{nominee; accountId; observers} <- fetchAndArchive openAllocationAccountRequestCid
-
-          let account = Account with provider; owner = customer; id = accountId
-          createOrLookup AllocationAccountRule with ..
-
-      nonconsuming CloseAccount : ()
-        with
-          closeAccountRequestCid : ContractId Custody.CloseAccountRequest
-        do
-          closeAccountRequest <- fetchAndArchive closeAccountRequestCid
-          (assetSettlementRuleCid, _) <- fetchByKey @AssetSettlementRule closeAccountRequest.accountId
-          archive assetSettlementRuleCid
-
-      nonconsuming CloseAllocationAccount : ()
+      nonconsuming Deposit : ContractId AssetDeposit
         with
-          closeAllocationAccountRequestCid : ContractId Custody.CloseAllocationAccountRequest
+          depositRequestCid : ContractId Custody.DepositRequest
         do
-          closeAccountRequest <- fetchAndArchive closeAllocationAccountRequestCid
-          (allocationAccountRuleCid, _) <- fetchByKey @AllocationAccountRule closeAccountRequest.accountId
-          archive allocationAccountRuleCid
+          Custody.DepositRequest{asset} <- fetchAndArchive depositRequestCid
+          (_, AssetDescription{registrar}) <- fetchByKey @AssetDescription asset.id
 
-      nonconsuming CreditAccount : ContractId AssetDeposit
-        with
-          creditAccountRequestCid : ContractId Custody.CreditAccountRequest
-        do
-          Custody.CreditAccountRequest{accountId; asset} <- fetchAndArchive creditAccountRequestCid
-          depositCid <- exerciseByKey @AssetSettlementRule accountId AssetSettlement_Credit with ctrl = provider; ..
-          deposit <- fetch depositCid
-          exercise depositCid AssetDeposit.AssetDeposit_SetObservers with newObservers = deposit.asset.id.signatories `union` deposit.observers
+          create AssetDeposit with asset; account; lockers = empty; observers = fromList [ registrar ]
 
```

Comment:
Shouldn't the observer here be the `issuer` of the asset, not the `registrar`? The registrar should already be the same as the custody provider on this service (and we should probably enforce this), and it would be a signatory on that asset deposit (and therefore see it).